### DISCUSSION TITLE
Is there a way to only access the new labelled data in dataset.label ?

### DISCUSSION THREAD

**Discussion by Karthik-Pandaram** [baal-org/baal#231] [Q&A] [answered]
Hi I wanted to know if there is a way to access the newly labelled data alone from the ```dataset object``` in the AL loop.
```
for i in range(5):
  
  print("Dataset size", len(dataset))
  clf.fit(*dataset.dataset)
  predictions = clf.predict(test_x)
  print("Test Accuracy  :: ", accuracy_score(test_y, predictions))
  probs = predict(dataset.pool, clf)
  to_label = heuristic(probs)
  query_size = 10

  if len(to_label) > 0:
      dataset.label(to_label[: query_size])
      break
  else:
    break
```
here the ```to_label``` contains the indexes to be labelled but when I try ```train_x.iloc[x]``` the results are different from the those in the ```dataset.dataset[0]```.
Can some one explain me how the dataset.label adds new samples to the labelled pool?

**Comment by Dref360** [comment, marked as answer]
Hello!

In this case, the indices in `to_label` are relative to the unlabelled pool, not to the overall dataset.

To get the mapping from the pool to the "original" indices, you can use `dataset._pool_to_oracle_index(to_label[: query_size])` **before** `dataset.label`.

**Addtional infos**:
* `ActiveLearningDataset` keeps track of what is labelled and what is not.
* `ActiveNumpyDataset.dataset` gives you a new array with labelled items only.
* `ActiveNumpyDataset.pool` gives you an array with unlabelled items only.
* `ActiveNumpyDataset.label(indices)` add items from the unlabelled pool to the labelled pool before adding these indices to the set of labelled items.


I hope this helps!

**Reply by Karthik-Pandaram** [reply]
Hi Thanks for the reply it is helpful !.

It would be great if we could have a method to get the newly labelled samples alone as a separate dataset, before adding to the pool.

**Reply by Dref360** [reply]
Hmm right, I think the easiest way is still `dataset.pool[0][to_label[:query_size]]` with `ActiveNumpyArray`, but I understand that it is a bit complicated.
